python/tank/descriptor/bundle_cache_usage/manager.py

```diff
@@ -0,0 +1,270 @@
+        scan the entire list for a link, symlink or such in the list. If found, we'll
+        abort the deletion process before actually deleting anything.
+
+        :param filelist: A list file and folders to be deleted
+        """
+
+        # First, check whether there is a symlink in the list
+        for f in filelist:
+            if os.path.islink(f):
+                # CAVEAT: Always False if symbolic links are not supported by the Python runtime.
+                #         How do we know whether it is supported???
+                raise BundleCacheUsageFileDeletionException(f, "Found a symlink")
+
+        # We have a crude list, now we need to sort it out in reverse
+        # order so we can later on delete files, and then parent folder
+        # in a logical order.
+        rlist = list(reversed(filelist))
+        # No symlinks, Houston we're clear for deletion
+        for f in rlist:
+            if not os.path.exists(f):
+                raise BundleCacheUsageFileDeletionException(
+                    f,
+                    "Attempting to delete non existing file or folder."
+                )
+
+            if os.path.isfile(f):
+                os.remove(f)
+
+            elif os.path.isdir(f):
+                # Because we're deleting items that should be reverse prdered
+                # when we're about to delete a folder, it should be empty already.
+                # let's check it out!
+                try:
+                    os.rmdir(f)
+                except OSError as e:
+                    raise BundleCacheUsageFileDeletionException(
+                        f,
+                        "Attempted to delete a non-empty folder: %s" % (e)
+                    )
+
+            else:
+                raise BundleCacheUsageFileDeletionException(
+                    f,
+                    "Not a link, not a file, not a directory ???"
+                )
+
+    ###################################################################################################################
+    #
+    # PUBLIC API - methods & properties
+    #
+    # Can be called from any threading contextes (main or worker)
+    #
+    ###################################################################################################################
+
+    @property
+    def bundle_cache_root(self):
+        return self._bundle_cache_root
+
+    def initial_populate(self):
+        """
+        Scan the bundle cache at the specified location for bundles and add them
+        as unused entries to the database.
+
+        """
+
+        log.info("Searching for existing bundles ...")
+        start_time = time.time()
+        found_bundles = self._find_bundles()
+        for bundle_path in found_bundles:
+            self._worker.add_unused_bundle(bundle_path)
+
+        log.info("populating done in %ss, found %d entries" % (
+            time.time() - start_time, len(found_bundles)
+        ))
+
+    def get_bundle_count(self):
```

Comment:
better if this is a property named `bundle_count`

Reply:
The way I usually decide between a property and method is based on whether the information is available right away or not. In this case, getting the number of bundle in the database requires an open database, a cursor context and a query counting the number of rows.  Secondly, for consistency with the other database related methods, I though getting a 'get_' type method for everyone was cleaner.

With that said, I have no problem changing that to a property if you really want that to be a property.